app/controllers/devise_otp/devise/otp_tokens_controller.rb

```diff
@@ -19,16 +19,24 @@ def show
         end
       end
 
+      #
+      # Displays the QR Code and Validation Token form for enabling the OTP
+      #
+      def edit
```

Comment:
Why not to populate the code here?

Reply:
Sure, we could definitely do that. I was just trying to follow the pre-populatino strategy via the "on create" callback within the model.

In that case, would we disable the "on create" callback in the model? It seems simpler to assume nobody has a pre-populated token so that all users are on a level playing field.

Reply:
I feel not having it populated is a bit nicer and we just do it here for users that are ready to use OTP

Reply:
OK, cool! I will try this route. Feel free to hold off on the responses below.

Reply:
Yes let's put it here so it doesn't need the extra action for older apps. I also think it's cleaner

Reply:
Done!